pkg/apis/serving/v1/revision_lifecycle.go

```diff
@@ -168,6 +168,26 @@ func (rs *RevisionStatus) PropagateDeploymentStatus(original *appsv1.DeploymentS
 	}
 }
 
+func (rs *RevisionStatus) PropagateDeploymentAvailabilityStatusIfFalse(original *appsv1.DeploymentStatus) {
+	m := revisionCondSet.Manage(rs)
+	avCond := m.GetCondition(RevisionConditionResourcesAvailable)
+
+	// Skip if set for other reasons
```

Comment:
What other reasons? Why would we need to skip in that case?

Reply:
I don't want to change the current state machine see comment: https://github.com/knative/serving/pull/14835#issuecomment-1910360909. So I am only targeting a specific case, if for example deployment faced an issue and revision avail condition is set to false already I am not going to update it and set it to false again because, I am just skipping the update. 
In general it should not make a difference as if that was an intermediate state to have avail cond false (it happens when replicas are not ready) for some other reason then it is going to be reset to true and then later we will set it to false anyway. I can test without this in a test PR but I am wondering for side effects.

Reply:
Ok, the I think the comment was just not explaining this fully.